types/node-forge/md.d.ts

```diff
@@ -0,0 +1,67 @@
+declare module "node-forge" {
+    const md: {
+        sha1: {
+            create(): md.sha1.MessageDigest;
+        };
+        sha256: {
+            create(): md.sha256.MessageDigest;
+        };
+        sha512: {
+            create<TAlg extends md.sha512.AlgorithmSelection = md.sha512.AlgorithmSelection.Sha512>(
+                /** @default 'SHA-512' */
+                algorithm?: TAlg,
+            ): TAlg extends md.sha512.AlgorithmSelection.Sha384 ? md.sha512.Sha384MessageDigest
+                : TAlg extends md.sha512.AlgorithmSelection.Sha512224 ? md.sha512.Sha512224MessageDigest
+                : TAlg extends md.sha512.AlgorithmSelection.Sha512256 ? md.sha512.Sha512256MessageDigest
+                : TAlg extends md.sha512.AlgorithmSelection.Sha512 ? md.sha512.Sha512MessageDigest
+                : never;
+            sha224: {
+                create(): md.sha512.Sha512224MessageDigest;
+            };
+            sha256: {
+                create(): md.sha512.Sha512256MessageDigest;
+            };
+            sha384: {
+                create(): md.sha512.Sha384MessageDigest;
+            };
+        };
+        sha384: typeof md.sha512.sha384;
+        "sha512/224": typeof md.sha512.sha224;
+        "sha512/256": typeof md.sha512.sha256;
+        md5: {
+            create(): md.md5.MessageDigest;
+        };
+        algorithms: {
+            md5: typeof md.md5;
+            sha1: typeof md.sha1;
+            sha256: typeof md.sha256;
+            sha384: typeof md.sha384;
+            sha512: typeof md.sha512;
+            "sha512/224": (typeof md)["sha512/224"];
+            "sha512/256": (typeof md)["sha512/256"];
+        };
+    };
+
+    const md5: typeof md.md5;
+    const sha1: typeof md.sha1;
+    const sha256: typeof md.sha256;
+    const sha512: typeof md.sha512 & {
+        sha384: typeof md.sha512.sha384;
+    };
+    const sha384: typeof md.sha512.sha384;
```

Comment:
This is not a pure move.

Reply:
It's this line: https://github.com/digitalbazaar/forge/blob/2bb97afb5058285ef09bcf1d04d6bd6b87cffd58/lib/sha512.js#L23

Reply:
I think that was already correctly accounted for by `md.sha384: typeof md.sha512.sha384`, and your added `&` is redundant.

But if there was a problem here to be addressed, do it in a separate commit.